sopel/irc/isupport.py

```diff
@@ -131,10 +131,38 @@ def _parse_prefix(value):
 }
 
 
+def _hex_unescape(param):
+    """
+    Unescape hexadecimal escape sequences (i.e. ``'\\xHH'``)
+
+    Sopel follows the recommendation of `modern.ircdocs.horse`__ which only
+    allows the escapes ``\\x20, \\x5C, \\x3D``, an exception will be raised
+    for any other escape sequence
+
+    .. __: https://modern.ircdocs.horse/#rplisupport-005
+    """
+    HEX_PATTERN = r"\\x([0-9a-fA-F]{2})"
+
+    def _unescape(match):
+        num = match.group(1).upper()
+        if num == "20":
+            result = " "
+        elif num == "5C":
+            result = "\\"
+        elif num == "3D":
+            result = "="
+        else:
+            raise ValueError("Encountered invalid escape \\x{num}".format(num=num))
```

Comment:
This shouldn't raise an exception: for instance, [inspircd will happily escape](https://github.com/inspircd/inspircd/blob/2b810b412aeee2d9c79d97fa54a80e317981a6d8/src/server.cpp#L181-L200) `\n` and replace it with `"\x0a"`, which is going to raise an exception here.

As a result, I suggest to do the following:

```suggestion
            result = match.group(0)
```